### PR TITLE
Bump Django to 1.11.17 and add Python 3.7 in test matrix

### PR DESCRIPTION
**Pull request by benjaoming**
### Summary

This adds Python 3.7 compatibility, assuming that it's confirmed with by CI...

### Reviewer guidance

Any discussion points about targeting this for 0.11.1?

### References

#4548

----

### Contributor Checklist

- [ ] Contributor has fully tested the PR manually
- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [x] If there are any front-end changes, before/after screenshots are included
- [x] If this is an important user-facing change, PR or related issue has a 'changelog' label

### Reviewer Checklist

- [ ] Automated test coverage is satisfactory
- [ ] Reviewer has fully tested the PR manually
- [ ] PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- [ ] External dependencies files were updated (`yarn` and `pip`)
- [ ] Documentation is updated
- [ ] Link to diff of internal dependency change is included
- [ ] CHANGELOG.rst is updated for high-level changes
- [ ] Contributor is in AUTHORS.rst

### CODE REVIEW

**Comment by indirectlylit** [comment]
unclear to me why tests are failing:

```
Traceback (most recent call last):
  File "/home/travis/build/learningequality/kolibri/.tox/py3.7/bin/kolibri", line 11, in <module>
    load_entry_point('kolibri', 'console_scripts', 'kolibri')()
  File "/home/travis/build/learningequality/kolibri/kolibri/utils/cli.py", line 642, in main
    manage(command, args=django_args)
  File "/home/travis/build/learningequality/kolibri/kolibri/utils/cli.py", line 432, in manage
    execute_from_command_line(argv=argv)
  File "/home/travis/build/learningequality/kolibri/.tox/py3.7/lib/python3.7/site-packages/django/core/management/__init__.py", line 364, in execute_from_command_line
    utility.execute()
  File "/home/travis/build/learningequality/kolibri/.tox/py3.7/lib/python3.7/site-packages/django/core/management/__init__.py", line 356, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/home/travis/build/learningequality/kolibri/.tox/py3.7/lib/python3.7/site-packages/django/core/management/base.py", line 283, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/home/travis/build/learningequality/kolibri/.tox/py3.7/lib/python3.7/site-packages/django/core/management/base.py", line 327, in execute
    self.check()
  File "/home/travis/build/learningequality/kolibri/.tox/py3.7/lib/python3.7/site-packages/django/core/management/base.py", line 359, in check
    include_deployment_checks=include_deployment_checks,
  File "/home/travis/build/learningequality/kolibri/.tox/py3.7/lib/python3.7/site-packages/django/core/management/base.py", line 346, in _run_checks
    return checks.run_checks(**kwargs)
  File "/home/travis/build/learningequality/kolibri/.tox/py3.7/lib/python3.7/site-packages/django/core/checks/registry.py", line 81, in run_checks
    new_errors = check(app_configs=app_configs)
  File "/home/travis/build/learningequality/kolibri/.tox/py3.7/lib/python3.7/site-packages/django/core/checks/urls.py", line 16, in check_url_config
    return check_resolver(resolver)
  File "/home/travis/build/learningequality/kolibri/.tox/py3.7/lib/python3.7/site-packages/django/core/checks/urls.py", line 26, in check_resolver
    return check_method()
  File "/home/travis/build/learningequality/kolibri/.tox/py3.7/lib/python3.7/site-packages/django/urls/resolvers.py", line 256, in check
    for pattern in self.url_patterns:
  File "/home/travis/build/learningequality/kolibri/.tox/py3.7/lib/python3.7/site-packages/django/utils/functional.py", line 35, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/home/travis/build/learningequality/kolibri/.tox/py3.7/lib/python3.7/site-packages/django/urls/resolvers.py", line 407, in url_patterns
    patterns = getattr(self.urlconf_module, "urlpatterns", self.urlconf_module)
  File "/home/travis/build/learningequality/kolibri/.tox/py3.7/lib/python3.7/site-packages/django/utils/functional.py", line 35, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/home/travis/build/learningequality/kolibri/.tox/py3.7/lib/python3.7/site-packages/django/urls/resolvers.py", line 400, in urlconf_module
    return import_module(self.urlconf_name)
  File "/home/travis/build/learningequality/kolibri/.tox/py3.7/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/travis/build/learningequality/kolibri/kolibri/deployment/default/urls.py", line 40, in <module>
    url(r'', include('kolibri.core.urls')),
  File "/home/travis/build/learningequality/kolibri/.tox/py3.7/lib/python3.7/site-packages/django/conf/urls/__init__.py", line 50, in include
    urlconf_module = import_module(urlconf_module)
  File "/home/travis/build/learningequality/kolibri/.tox/py3.7/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/travis/build/learningequality/kolibri/kolibri/core/urls.py", line 56, in <module>
    url(r'^api/', include('kolibri.core.api_urls')),
  File "/home/travis/build/learningequality/kolibri/.tox/py3.7/lib/python3.7/site-packages/django/conf/urls/__init__.py", line 50, in include
    urlconf_module = import_module(urlconf_module)
  File "/home/travis/build/learningequality/kolibri/.tox/py3.7/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/travis/build/learningequality/kolibri/kolibri/core/api_urls.py", line 8, in <module>
    url(r'^tasks/', include('kolibri.core.tasks.api_urls')),
  File "/home/travis/build/learningequality/kolibri/.tox/py3.7/lib/python3.7/site-packages/django/conf/urls/__init__.py", line 50, in include
    urlconf_module = import_module(urlconf_module)
  File "/home/travis/build/learningequality/kolibri/.tox/py3.7/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/travis/build/learningequality/kolibri/kolibri/core/tasks/api_urls.py", line 3, in <module>
    from .api import TasksViewSet
  File "/home/travis/build/learningequality/kolibri/kolibri/core/tasks/api.py", line 7, in <module>
    from iceqube.common.classes import State
  File "/home/travis/build/learningequality/kolibri/.tox/py3.7/lib/python3.7/site-packages/iceqube/common/classes.py", line 5, in <module>
    from iceqube.common.utils import import_stringified_func, stringify_func
  File "/home/travis/build/learningequality/kolibri/.tox/py3.7/lib/python3.7/site-packages/iceqube/common/utils.py", line 9
    from iceqube import async
                            ^
```

**Comment by benjaoming** [comment]
@indirectlylit it's because of `async`, its a syntax keyword now.

https://github.com/learningequality/iceqube/pull/34

https://docs.python.org/3/library/asyncio.html

**Comment by benjaoming** [comment]
Doing some work for the Iceqube CI, so we can have automated tests approve these changes...

**Comment by benjaoming** [comment]
This PR is pending that I have credentials for PyPi to release another Iceqube...

**Comment by rtibbles** [comment]
Hrm, sorry, I don't have creds for icequebe on pypi either.

**Comment by benjaoming** [comment]
Iceqube 0.1b1 is released!

**Comment by benjaoming** [comment]
What what what. @lyw07 have you seen this before?

```
python build_tools/install_cexts.py --file "requirements/cext.txt" # pip install c extensions
  Could not find a version that satisfies the requirement cffi!=1.11.3,>=1.7 (from cryptography==2.3) (from versions: 0.7.1, 0.8, 0.8.4, 1.1.1, 1.1.2, 1.4.1, 1.5.2)
No matching distribution found for cffi!=1.11.3,>=1.7 (from cryptography==2.3)
 
Download failed for package cryptography-2.3-cp36-cp36m-linux_armv7l.whl.
```

**Comment by lyw07** [comment]
Ah yes. I just created a PR(https://github.com/learningequality/kolibri/pull/4678) targeting 0.9 to fix it. Will get the fix in to branches above 0.9 after the PR is merged. 
The error is because in the arm wheel repository we use, they have cryptography py36 but don’t have cffi py36. Their build for cffi py36 might be failed. I have commented on their github issue to ask about it. But before that, I think the workaround would be to skip downloading python 3.6 Linux arm cryptography package..

**Comment by benjaoming** [comment]
Thanks so much, that makes total sense -- sorry haven't familiarized myself with all the latest work of 2019 :)

**Comment by benjaoming** [comment]
There are a handful of Python 3.7 related issues in tests that I will investigate, and the issue with the missing ARM build for Cryptography isn't really a blocker for fixing this.

I'll try and dig into it and assess if its feasible to fix for 0.11.1 -- and if not, retargetting this PR is easy.

File: .travis.yml
Comment (rtibbles) [inline .travis.yml:37]: Is this a travis requirement to run with sudo true for Python3.7?

File: kolibri/core/content/templatetags/content_tags.py
Comment (rtibbles) [inline kolibri/core/content/templatetags/content_tags.py:37]: Sorry.... I was obviously not clairvoyant enough when coming up with these kwargs.

File: kolibri/core/webpack/utils.py
Comment (rtibbles) [inline kolibri/core/webpack/utils.py]: Good good - yeah, maybe add a deprecation warning if `kwargs.pop('async')` returns something?

File: kolibri/plugins/coach/templatetags/coach_tags.py
Comment (rtibbles) [inline kolibri/plugins/coach/templatetags/coach_tags.py:10]: I thought we had already linted every file? When we run CI linting, does it not run our import linting too?

**Review by rtibbles — COMMENTED** [review]
This is looking good, and I think would be good to include in 0.11.1 if at all possible.

File: .travis.yml
Comment (benjaoming) [reply .travis.yml:37]: Yes, it's because it obtains Python 3.7 from some addition source and installs it. We cannot specify "Python 3.7" without these extra options. It's been the case for quite some time now, but Travis seems to need some time for releasing new images.

File: kolibri/core/content/templatetags/content_tags.py
Comment (benjaoming) [reply kolibri/core/content/templatetags/content_tags.py:37]: I think it would have been caught by linting, but we have only linted in Python 2 thus far.

Comment (rtibbles) [reply kolibri/core/content/templatetags/content_tags.py:37]: Way to fix that!

File: kolibri/core/webpack/utils.py
Comment (benjaoming) [reply kolibri/core/webpack/utils.py]: Good point, let's have them visible! Fixed!

File: kolibri/core/content/templatetags/content_tags.py
Comment (benjaoming) [reply kolibri/core/content/templatetags/content_tags.py:37]: Yes also additional note here: AFAIK, flake8 is linting for the active Python version. I have configured it just now to run on Python 3.6 to use the supposedly slightly faster Travis image.

From http://flake8.pycqa.org/en/latest/

> It is very important to install Flake8 on the correct version of Python for your needs. If you want Flake8 to properly parse new language features in Python 3.5 (for example), you need it to be installed on 3.5 for Flake8 to understand those features. In many ways, Flake8 is tied to the version of Python on which it runs.

File: kolibri/plugins/coach/templatetags/coach_tags.py
Comment (benjaoming) [reply kolibri/plugins/coach/templatetags/coach_tags.py:10]: This may not be seen by a linting rule but is likely only caused by changes to our pre-commit hooks.

**Review by rtibbles — APPROVED** [review]
Mr. Travis will have the final say, but this gets my vote!

**Comment by rtibbles** [comment]
Two tiny linting errors on the Python3 lint.

**Comment by benjaoming** [comment]
Yup! CC: @radinamatic - we could need a bit of Python 3.7 testing before announcing the support. I'll have a look at building a Debian package that doesn't force any Python downgrades.

**Pull request merged**
